tests/DTS/Repository/TemplateRepositoryTest.php

```diff
@@ -17,22 +19,28 @@ class TemplateRepositoryTest extends TestCase {
   /** @var TemplatePersistence */
   private $p;
 
+  /** @var CompiledTemplatePersistence */
+  private $ctp;
+
   /** @var TemplateRepository */
   private $repo;
 
   private $docType;
 
-  public function setUp(): void {
+  protected function setUp(): void {
     parent::setUp();
     $this->db = Db::connection();
+    $this->db->beginTransaction();
+
     $this->p = new TemplatePersistence($this->db);
-    $this->repo = new TemplateRepository($this->p);
+    $this->ctp = new CompiledTemplatePersistence($this->db);
+    $this->repo = new TemplateRepository($this->p, $this->ctp);
     $this->docType = 'TemplateRepositoryTest-' . time();
   }
 
-  public function tearDown(): void {
-    $this->db->deleteWhere('template', ['doc_type' => $this->docType]);
+  protected function tearDown(): void {
     parent::tearDown();
+    $this->db->rollBack();
```

Comment:
strictly speaking for proper nesting you should put your tear down above the parent tear down